### PR TITLE
Feat/iss 19: Configure Husky with ESLint, Prettier, and Commitlint in pre-commit

### PR DESCRIPTION
**Pull request by joset98**
### Description

This PR adds the configuration of Husky to run ESLint, Prettier, and Commitlint in the pre-commit, ensuring that all code committed is formatted and adheres to the established commit conventions.

- Installed Husky, lint-staged, @commitlint/config-conventional and @commitlint/cli.
- Created pre-commit hook to run lint-staged.
- Created commitlint.config.js file with conventional commit configuration.
- Created commit-msg hook to validate commit messages.

Fixes:
 - #19 

### How to test it

Make changes to any file in the project. Try committing and verify that Husky runs ESLint, Prettier, and Commitlint before allowing the commit.

### Checklist
 This issue can be closed when the following tasks are complete:
 
 - [x] Husky run in pre-commit EsLint, Prettier and Conventional Commit

### Screenshots

>image

>image

### CODE REVIEW

File: .husky/commit-msg [hunk 1]
```diff
@@ -0,0 +1 @@
+npx --no -- commitlint --edit $1
```
Comment (Israel-Laguan) [inline .husky/commit-msg:1]: Sorry, what is this for?

File: .lintstagedrc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "*/**/*.{js,ts}": ["prettier --write", "eslint --fix", "eslint"],
```
Comment (Israel-Laguan) [inline .lintstagedrc:2]: You can remove the ts

File: package.json [hunk 3]
```diff
@@ -8,7 +8,8 @@
     "lint": "eslint .",
     "lint:fix": "eslint . --fix",
     "format": "prettier --write \"**/*.{js,jsx,json,md}\"",
-    "format:check": "prettier --check \"**/*.{js,jsx,json,md}\""
+    "format:check": "prettier --check \"**/*.{js,jsx,json,md}\"",
```
Comment (Israel-Laguan) [inline package.json:11]: You can remove jsx

File: .husky/commit-msg (hunk 1 above)
Comment (joset98) [reply .husky/commit-msg:1]: it comes from docs https://commitlint.js.org/guides/local-setup.html:

>yarn add --dev husky

>yarn husky init

> # Add commit message linting to commit-msg hook
> echo "yarn commitlint --edit \$1" > .husky/commit-msg

Comment (joset98) [reply .husky/commit-msg:1]: the `--no --` part ensures that commitlint runs without any global npm configurations interfering, and --edit $1 passes the commit message file to commitlint for validation.

**Review by heliomar-pena — APPROVED** [review]
LGTM. thanks!

**Pull request merged**